devel/libenkf/src/enkf_main.c

```diff
@@ -770,17 +768,50 @@ static int enkf_main_serialize_dataset( const ensemble_config_type * ens_config
         enkf_main_serialize_node( key , active_list , row_offset[ikw] , work_pool , serialize_info );
         current_row += active_size[ikw];
       }
-      const module_data_block_type * data_block = module_data_block_alloc( key, active_list_get_active(active_list),  row_offset[ikw], active_size[ikw] );
-      module_data_block_vector_add_data_block(module_data_block_vector, data_block);
     }
   }
   matrix_shrink_header( A , current_row , ens_size );
   stringlist_free( update_keys );
   return matrix_get_rows( A );
 }
 
+static void enkf_main_init_module_info( const obs_data_type * obs_data,
```

Comment:
This function does not seem to be in use at all? If possible I would prefer if [RAII](https://en.wikipedia.org/wiki/Resource_Acquisition_Is_Initialization) could be used - i.e.
```C
module_info_type * enkf_main_alloc_module_info( ... )  {
    // Create and initialize the module_info instance.
} 
```